### PR TITLE
[UC04#34] Implemented RPC Signin() that allows visitor to login as registered user

### LINKED ISSUE #34: Creare RPC SignIn [UC04]
_**Dod**_
- [x]  La pull request ha un titolo e descrizione adeguata
- [x]  La pull request è stata collegata con la issue del task
- [x]  I requisiti funzionali del task collegato sono soddisfatti
- [x]  Il codice si compila senza errori
- [x]  Gli Unit Tests sono stati scritti e passati 
- [x]  Gli Unit Tests esistenti sono stati passati
- [x]  La documentazione è stata aggiornata
- [x]  Il Code Review è stato fatto
- [x]  La pull request è stata accettata e fatto merge sul branch development.

 **Effort effettivo:**

### PR DESCRIPTION
**Pull request by alessioarcara**
This PR implements #34, adding the necessary RPC for visitors to login as registered users through the application's user interface in future updates. It also includes validation for incorrectly formatted emails and passwords with less than 8 characters. Lastly, a slight update has been made to the login interface. 

Supplied with coverage for unit testing

### CODE REVIEW

File: src/test/java/com/aadm/cardexchange/server/AuthServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+        Assertions.assertThrows(IllegalArgumentException.class, () -> authService.signup("", "password"));
+    }
+
+    @Test
+    public void testSignupForIncorrectEmails() {
+        Assertions.assertAll(() -> {
+            Assertions.assertThrows(IllegalArgumentException.class, () -> authService.signup("test", "password"));
+            Assertions.assertThrows(IllegalArgumentException.class, () -> authService.signup("test@", "password"));
+            Assertions.assertThrows(IllegalArgumentException.class, () -> authService.signup("test@test", "password"));
+            Assertions.assertThrows(IllegalArgumentException.class, () -> authService.signup("test@test.", "password"));
```
Comment (Matte0Sacc0) [inline src/test/java/com/aadm/cardexchange/server/AuthServiceTest.java:53]: Please add a test for an email and a password whit a space character inside

File: src/main/java/com/aadm/cardexchange/server/AuthServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    public AuthServiceImpl() {
+        db = new MapDBImpl();
+    }
+
+    public AuthServiceImpl(MapDB mockDB) {
+        db = mockDB;
+    }
+
+    private boolean validateEmail(String email) {
+        Matcher matcher = emailPattern.matcher(email);
```
Comment (Matte0Sacc0) [inline src/main/java/com/aadm/cardexchange/server/AuthServiceImpl.java:34]: As said in the previous PR I suggest to handle also the typo of blank spaces in the strings of email and password. It only consists in using the function .trim() so it should take few minutes

**Review by Matte0Sacc0 — COMMENTED** [review]
Same as the previous, is everything written in the code comments

**Comment by alessioarcara** [comment]
The regex included in this PR already addresses the issue described

**Pull request merged**